lib/FreeRTOS-Plus-TCP/source/FreeRTOS_DNS.c

```diff
@@ -809,25 +824,14 @@ static uint8_t *prvSkipNameField( uint8_t *pucByte, size_t xSourceLen )
 
 uint32_t ulDNSHandlePacket( NetworkBufferDescriptor_t *pxNetworkBuffer )
 {
-uint8_t *pucUDPPayloadBuffer;
-size_t xPlayloadBufferLength;
 DNSMessage_t *pxDNSMessageHeader;
 
-	xPlayloadBufferLength = pxNetworkBuffer->xDataLength - sizeof( UDPPacket_t );
-	if ( xPlayloadBufferLength < sizeof( DNSMessage_t ) )
-	{
-		return pdFAIL;
-	}
+	pxDNSMessageHeader =
+		( DNSMessage_t * )( pxNetworkBuffer->pucEthernetBuffer + sizeof( UDPPacket_t ) );
 
-	pucUDPPayloadBuffer = pxNetworkBuffer->pucEthernetBuffer + sizeof( UDPPacket_t );
-	pxDNSMessageHeader = ( DNSMessage_t * ) pucUDPPayloadBuffer;
-
-	if( pxNetworkBuffer->xDataLength > sizeof( UDPPacket_t ) )
-	{
-		prvParseDNSReply( pucUDPPayloadBuffer,
-			xPlayloadBufferLength,
-			( uint32_t )pxDNSMessageHeader->usIdentifier );
-	}
+	prvParseDNSReply( ( uint8_t * )pxDNSMessageHeader,
+					  pxNetworkBuffer->xDataLength,
+					  ( uint32_t )0u );
```

Comment:
What would be the rationale for returning a DNS result to the app when the packet didn't match our (cryptographically random) transaction ID? Seems to me that it's not just a question of whether we cache the result.

Reply:
The above call is done for packets that come in asynchronously. `prvParseDNSReply()` will look-up the transaction ID in the list of outstanding DNS requests. Only if a match is found, the result will be added to the DNS cache.